lib/synapse/config_generator.rb

```diff
@@ -0,0 +1,20 @@
+require 'synapse/log'
+require 'synapse/config_generator/base'
+
+module Synapse
+  class ConfigGenerator
+    # the type which actually dispatches generator creation requests
+    def self.create(type, opts)
+      generator = begin
+        type = type.downcase
+        require "synapse/config_generator/#{type}"
```

Comment:
I was under the impression this is already possible if you install via gems? All you have to do is put your class in the right path and then synapse dynamically requires it, or that's what I thought the design from Nerve allowed you to do.

I can whip up a quick example gem to see if this is flawed thinking, but I thought that was why we switched to this way of constructing classes in Nerve (and Synapse).